components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1037,8 +1037,17 @@ private class MonitoredItemWrapper {
                 if (DataTemplate != null) {
                     Item.Filter = DataTemplate.DataChangeFilter.ToStackModel() ??
                         ((MonitoringFilter)DataTemplate.AggregateFilter.ToStackModel(session.MessageContext));
-                } else if (EventTemplate != null) {
-                    Item.Filter = codec.Decode(EventTemplate.EventFilter, true);
+                }
+                else if (EventTemplate != null) {
+                    var eventFilter = codec.Decode(EventTemplate.EventFilter, true);
+                    if (EventTemplate.PendingAlarm != null && EventTemplate.PendingAlarm.Enabled) {
+                        eventFilter.SelectClauses.Add(new SimpleAttributeOperand() {
```

Comment:
This could have been added I think manually as well? Do we have to check first if it's already there before we add a SimpleAttributeOperand like this?

Reply:
Fixed, good point